src/index.js

```diff
@@ -11,12 +11,15 @@ const editor = document.getElementById('editor');
 const comment = document.getElementById('comment');
 const output = document.getElementById('output');
 const context = output.getContext('2d');
+const dpr = window.devicePixelRatio || 1;
 
 let callback = function() {};
 let startTime = new Date();
 let code = '';
 
-output.width = output.height = width;
+output.width = output.height = width * dpr;
+context.scale(dpr,dpr);
+output.style.width = output.style.height = `${width}px`;
```

Comment:
**TL;DR:** We should skip this line, and instead size the canvas and form elements using `vmin` or similar viewport unit.

With this change the size of the canvas will adapt to the DPR but not the actual viewport size. Furthermore, it can end up having a different size than the labels and form inputs below.

Reply:
Here's the type of CSS change I mean,

**style.styl**
```
html {
  margin: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 90%;
  background: #000;
  color: #fff;
}

body {
  margin: 0;
  width: 64vmin;
  height: 76vmin;
}

* {
  font-size: 2vmin;
  font-family: monospace;
  line-height: 2vmin;
}

canvas,form,input,label {
  padding: 0;
  background: transparent;
  width: 64vmin;
  display: block;
  margin: 0 auto;
  color: inherit;
  border: none; 
}

#output {
  height: 64vmin;
  margin-bottom: 4vmin;
  background: #222;
}

#comment * {
  color: #F24;
}
```

Reply:
I think these are great suggestions, but probably for a subsequent PR

Reply:
Fair
I filed the PR #19 ;)